astropy/coordinates/builtin_frames/icrs_cirs_transforms.py

```diff
@@ -51,11 +48,8 @@ def icrs_to_cirs(icrs_coo, cirs_frame):
         astrom_eb = CartesianRepresentation(astrom['eb'], unit=u.au,
                                             xyz_axis=-1, copy=False)
         newcart = icrs_coo.cartesian - astrom_eb
-
         srepr = newcart.represent_as(SphericalRepresentation)
-        i_ra = srepr.lon.to_value(u.radian)
-        i_dec = srepr.lat.to_value(u.radian)
-        cirs_ra, cirs_dec = atciqz(i_ra, i_dec, astrom)
+        cirs_ra, cirs_dec = atciqz(newcart.without_differentials(), astrom)
 
         newrep = SphericalRepresentation(lat=u.Quantity(cirs_dec, u.radian, copy=False),
                                          lon=u.Quantity(cirs_ra, u.radian, copy=False),
```

Comment:
If, on the line below, we use `distance=newcart.norm()`, we can get rid of `srepr` altogether here.